response-builder.js

```diff
@@ -213,7 +229,7 @@ class RichResponse {
     if (this.items.length > 0 && (this.items[0].basicCard ||
       this.items[0].structuredResponse)) {
       this.items.unshift(simpleResponseObj);
-    } else {
+    } else if (simpleResponseObj.simpleResponse) {
```

Comment:
Is this extra check necessary? Just wondering if there was a specific reason to add this check.

Reply:
I made that to just include a followupEvent without a simple response.
But as @smishra2 said, followupEvents has a little to do with rich responses. Moreover when dialogflow ignores speech and other things when they see a followUpEvent. But I've made in this way so you can have a way to go because maybe in the future events and speech can be coexist at same time. So I was delegating the problem of handler a complex response to dialogflow, not to developers.